Dockerfile

```diff
@@ -25,6 +30,316 @@ RUN pacman -Syu \
   texlive-bin \
   wget \
   --noconfirm
+  
+###############################################################################
+# AUR DEPENDENCIES
+###############################################################################
+
+# we need opam, perl-critic, verilator
+
+# package (sub) dependencies
+RUN pacman -Syu \
+  base-devel \
+  bison \
+  boost \
+  cmake \
+  ocaml \
+  perl-class-inspector \
+  perl-clone \
+  perl-config-tiny \
+  perl-email-address \
+  perl-exception-class \
+  perl-file-remove \
+  perl-file-sharedir \
+  perl-file-which \
+  perl-io-string \
+  perl-list-moreutils \
+  perl-module-build \
+  perl-module-pluggable \
+  perl-module-runtime \
+  perl-params-util \
+  perl-readonly \
+  perl-role-tiny \
+  perl-sub-exporter \
+  perl-test-deep \
+  perl-test-fatal \
+  perl-test-pod \
+  perl-test-pod-coverage \
+  perl-tidy \
+  re2c \
+  scons \
+  --noconfirm
+  
+# create build user and group 
+RUN groupadd -r build
+RUN useradd -r -g build build
+RUN mkdir /tmp/build
+
+# ensure permissions
+RUN chown -R build:build /tmp/build && \
+    chmod -R 771 /tmp/build
+
+# opam and its dependencies, DO NOT CHANGE ORDER! 
+#************************************************
+
+# build opam dependency (clasp)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/clasp.git
+WORKDIR /tmp/build/clasp
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/clasp
+RUN pacman --noconfirm -U *.tar.xz
+
+# build opam dependency (gringo)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/gringo.git
+WORKDIR /tmp/build/gringo
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/gringo
+RUN pacman --noconfirm -U *.tar.xz
+
+# build opam dependency (aspcud)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/aspcud.git
+WORKDIR /tmp/build/aspcud
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/aspcud
+RUN pacman --noconfirm -U *.tar.xz
+
+# build (opam)
+USER build 
+WORKDIR /tmp/build
+RUN git clone https://aur.archlinux.org/opam.git
+WORKDIR /tmp/build/opam
+RUN makepkg --noconfirm
+USER root 
+WORKDIR /tmp/build/opam
+RUN pacman --noconfirm -U *.tar.xz
```

Comment:
Honestly my problem is:
If I run yaourt to fetch the packages on which I can later run makepkg, I 
can either:

a) Call it on multiple packages to let if figure out which dependencies I 
need. This gives me no information about the order I need to install the 
packages in and is therefore rather useless.

b)  Call it on each separate package which makes it useless again because I 
need to figure out the packages and order on my own.

In the end I need to sit down either way, browse the packages online and 
figure out the dependency graph. Which sucks. I don't think I will ever use 
Gentoo.